app/ide-desktop/eslint.config.js

```diff
@@ -0,0 +1,354 @@
+            // Important to warn on accidental duplicated `interface`s e.g. when writing API wrappers.
+            '@typescript-eslint/no-redeclare': 'error',
+            'no-shadow': 'off',
+            '@typescript-eslint/no-shadow': 'warn',
+            'no-unused-expressions': 'off',
+            '@typescript-eslint/no-unused-expressions': 'error',
+            'jsdoc/require-param-type': 'off',
+            'jsdoc/check-access': 'warn',
+            'jsdoc/check-alignment': 'warn',
+            'jsdoc/check-indentation': 'warn',
+            'jsdoc/check-line-alignment': 'warn',
+            'jsdoc/check-param-names': 'warn',
+            'jsdoc/check-property-names': 'warn',
+            'jsdoc/check-syntax': 'warn',
+            'jsdoc/check-tag-names': 'warn',
+            'jsdoc/check-types': 'warn',
+            'jsdoc/check-values': 'warn',
+            'jsdoc/empty-tags': 'warn',
+            'jsdoc/implements-on-classes': 'warn',
+            'jsdoc/no-bad-blocks': 'warn',
+            'jsdoc/no-defaults': 'warn',
+            'jsdoc/no-multi-asterisks': 'warn',
+            'jsdoc/no-types': 'warn',
+            'jsdoc/no-undefined-types': 'warn',
+            'jsdoc/require-asterisk-prefix': 'warn',
+            'jsdoc/require-description': 'warn',
+            'jsdoc/require-description-complete-sentence': 'warn',
+            'jsdoc/require-file-overview': 'warn',
+            'jsdoc/require-hyphen-before-param-description': 'warn',
+            'jsdoc/require-param-description': 'warn',
+            'jsdoc/require-param-name': 'warn',
+            'jsdoc/require-property': 'warn',
+            'jsdoc/require-property-description': 'warn',
+            'jsdoc/require-property-name': 'warn',
+            'jsdoc/require-property-type': 'warn',
+            'jsdoc/require-returns-check': 'warn',
+            'jsdoc/require-returns-description': 'warn',
+            'jsdoc/require-throws': 'warn',
+            'jsdoc/require-yields': 'warn',
+            'jsdoc/require-yields-check': 'warn',
+            'jsdoc/tag-lines': 'warn',
+            'jsdoc/valid-types': 'warn',
+        },
+    },
+    {
+        files: ['**/*.js', '**/*.jsx', '**/*.cjs', '**/*.mjs'],
+        rules: {
+            '@typescript-eslint/no-var-requires': 'off',
+            // Parameter types must be specified using JSDoc in JS files.
+            'jsdoc/no-types': 'off',
+        },
+    },
+    {
+        files: ['**/*.ts', '**/*.mts', '**/*.cts', '**/*.tsx', '**/*.mtsx', '**/*.ctsx'],
+        ignores: ['**/*.d.ts'],
+        rules: {
+            'no-restricted-syntax': [
+                'error',
+                ...RESTRICTED_SYNTAXES,
+                {
+                    selector: '[declare=true]',
+                    message: 'No ambient declarations',
+                },
+            ],
+            // This rule does not work with TypeScript, and TypeScript already does this.
+            'no-undef': 'off',
+        },
+    },
+    {
```

Comment:
Oh wow, this is the biggest linter config I've seen ever :D Ehh, JavaScript world :P